TestRunner.java

```diff
@@ -0,0 +1,33 @@
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import java.util.*;
+
+public class TestRunner {
+    public static void main(String[] args) {
+        Result result = JUnitCore.runClasses(MainTest.class);
+
+        // Construct JSON output manually
+        StringBuilder json = new StringBuilder();
+        json.append("{");
+        json.append("\"state\": \"").append(result.getFailureCount() == 0 ? "Passed" : "Failed").append("\",");
```

Comment:
let's consistently use all-lowercase values (i.e. `"passed"` and `"failed"` instead of a mix of those and `"Passed"` and `"Failed"`) for the sake of consistency.